### PR TITLE
Check all MD files for deadlinks

### PR DESCRIPTION
**Pull request by bearni95**
- Closes ["check all MD files for deadlinks"](https://app.asana.com/0/1204649083736950/1204804605821065/f)
- [x] Detect broken links inside MD/MDX files
- [x] Detect broken links inside TSX files

### CODE REVIEW

File: packages/apps/docs/.gitignore [hunk 1]
```diff
@@ -17,3 +17,4 @@ src/pages/docs/chainweb/js-bindings
 
 next-env.d.ts
 
+broken-links.md
```
Comment (sstraatemans) [inline packages/apps/docs/.gitignore:20]: not needed when we are not saving it in a MD file

File: packages/apps/docs/scripts/detect-broken-links.js
Comment (sstraatemans) [inline packages/apps/docs/scripts/detect-broken-links.js:1]: can you move the file to ./src/scripts/detectBrokenLinks.mjs
and use imports?

File: packages/apps/docs/scripts/detect-broken-links.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        reportMd.push(`- ${link}`)
+      }
+    }
+
+    fs.writeFileSync(path.join(__dirname, '../broken-links.md'), reportMd.join('\n'))
+    console.log('Wrote broken links report to broken-links.md')
+  }  
+
+  if (Object.keys(filesWithBrokenLinks).length > 0) {
+    throw new Error('Found broken links:' + JSON.stringify(filesWithBrokenLinks, null, 2))
```
Comment (sstraatemans) [inline packages/apps/docs/scripts/detect-broken-links.js:167]: 👍🏻

File: packages/apps/docs/scripts/detect-broken-links.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    const reportMd = ['# Broken links']
+    for (const filePath in filesWithBrokenLinks) {
+      const links = filesWithBrokenLinks[filePath]
+      reportMd.push(`## ${filePath}`)
+      for (const link of links) {
+        reportMd.push(`- ${link}`)
+      }
+    }
+
+    fs.writeFileSync(path.join(__dirname, '../broken-links.md'), reportMd.join('\n'))
```
Comment (sstraatemans) [inline packages/apps/docs/scripts/detect-broken-links.js:162]: not needed. we want the build to just fail

File: packages/apps/docs/scripts/detect-broken-links.js
Comment (sstraatemans) [inline packages/apps/docs/scripts/detect-broken-links.js:1]: maybe we remove the external links? as you said it was way to slow?
Or is it ok and we let it in?

File: packages/apps/docs/scripts/detect-broken-links.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    if (link.startsWith('http')) {
+      if (!externalLinks[filePath]) {
+        externalLinks[filePath] = []
+      }
+
+      externalLinks[filePath].push(link)
+      return;
+    }
+
+    if (link.startsWith('/assets')) {
```
Comment (sstraatemans) [inline packages/apps/docs/scripts/detect-broken-links.js:26]: asset links can be absolute. so `/assets/blog/FILENAME`
so `../public......` now fails, but the file exists

File: packages/apps/docs/.gitignore (hunk 1 above)
Comment (bearni95) [reply packages/apps/docs/.gitignore:20]: the file's only generated when the optional flag `--report-md` is present

Comment (bearni95) [reply packages/apps/docs/.gitignore:20]: without a need for the md reports this should be gone

**Review by sstraatemans — APPROVED** [review]

**Pull request merged**